src/main/java/uk/gov/companieshouse/mapper/chips/ChipsFormDataMapper.java

```diff
@@ -102,6 +110,12 @@ private ChipsOfficer mapToOfficer(DissolutionDirector director) {
 
         officer.setPersonName(mapToPersonName(director.getName()));
         officer.setSignDate(asDateString(director.getDirectorApproval().getDateTime()));
+        officer.setEmail(director.getEmail());
+        officer.setIpAddress(director.getDirectorApproval().getIpAddress());
+
+        if (director.getOnBehalfName() != null) {
```

Comment:
You can do this to avoid having to invoke the getter twice:
```
Optional.ofNullable(director.getOnBehalfName()).ifPresent(officer::setOnBehalfName);
```

Reply:
Good idea